src/health-check/main.cpp

```diff
@@ -132,16 +132,14 @@ class HealthCheckerProcess : public ProtobufProcess<HealthCheckerProcess>
   {
     VLOG(1) << "Check passed";
 
-    // Send a healthy status update on the first success,
-    // and on the first success following failure(s).
-    if (initializing || consecutiveFailures > 0) {
-      TaskHealthStatus taskHealthStatus;
```

Comment:
>1) let the framework persist task health status before acknowledging the update
This approach looks better. @brat002 May you mind close this pull request and let me create a ticket in https://issues.apache.org/jira/browse/MESOS to track this?